.github/workflows/ci.yml

```diff
@@ -12,6 +12,7 @@ jobs:
       - run: npm run lint
       - run: npm run test
       - name: Detect Generation needed Dist
+        if: github.actor != 'dependabot'
```

Comment:
are you sure this is the name the actor gets when it runs as dependabot? I can think of more exceptions though. not all files end up generating a different dist folder contents. say changes to readme or anything else that is not source code really. are we sure it's safe to break the build just for the sake of somebody not running husky hooks properly? I'm not convinced honestly.

Reply:
I am +1 for simoneb stance, as a `github action` release need manually change the `package.json` version. It is not a good idea updating `dist` file without the change of package version.

Reply:
for example hooks do not run on my env (some projects running in WSL makes it crash), it's more convinient to ensure this in the CI. See: https://github.com/fastify/github-action-merge-dependabot/pull/61#pullrequestreview-725107329

Regarding dependabot AFAIK this is its account name.

Changes not in source code won't change the build artifact. On the other hand, dependabot will not run the `build` command and in the case of a dependency change we'll have to run it on our own.

The easiest thing is to run this on commit to default but like explain it can be a pain. Otherwise we can ship a Dockerfile which would be consumed with the build inside of it, but for our case it's kind of overkill.

Reply:
> I am +1 for simoneb stance. In addition, as a `github action` release need manually change the `package.json` version. It is not a good idea to force updating `dist` file without the change of package version.

A lot of people rely a non tagged actions, which means checking out the `default` branch. So in this case it means you're adding commits to the history without updating the code that the action will run. In the end the artifact of the default branch will not be in sync with the code.